raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -703,6 +704,37 @@ def set_payee_state_and_check_reveal_order(  # pylint: disable=invalid-name
     return list()
 
 
+def set_payee_state_and_check_reveal_order2(  # pylint: disable=invalid-name
+        transfers_pair,
+        payee_address,
+        new_payee_state):
+    """ Set the state of a transfer *sent* to a payee and check the secret is
+    being revealed backwards.
+    Note:
+        The elements from transfers_pair are changed in place, the list must
+        contain all the known transfers to properly check reveal order.
+    """
+    assert new_payee_state in MediationPairState.valid_payee_states
+
+    wrong_reveal_order = False
+    for back in reversed(transfers_pair):
+        if back.payee_address == payee_address:
+            back.payee_state = new_payee_state
+            break
+
+        elif back.payee_state not in STATE_SECRET_KNOWN:
+            wrong_reveal_order = True
+
+    if wrong_reveal_order:
+        # TODO: Append an event for byzantine behavior.
+        # XXX: With the current events_for_withdraw implementation this may
```

Comment:
I know that this is the same as the non `2` version of the function but I can not understand what it is trying to say. Can you maybe explain it a bit better and then once I can understand if we can edit the comment?

Reply:
the reveal secret message flows backwards, from the target to the initiator, if a node in the middle of the chain learns the secret faster something may be wrong, that's why there is a check for the reveal order.

things get slightly more complicated once one of the parties goes on-chain to unlock the lock, after that the secret is public and nodes may learn it directly from the blockchain, this means the check is not good anymore.

the comment is just a reminder about this

Reply:
All right. Cleaned the comment up a bit to make sense for me too as your explanation here in Github is better.